conjure-core/src/main/java/com/palantir/conjure/defs/validator/SafetyValidator.java

```diff
@@ -183,6 +217,18 @@ public Stream<String> visitReference(TypeName value) {
 
         @Override
         public Stream<String> visitExternal(ExternalReference value) {
+            String importType = getImportType(value);
+            if (ValidImportsForSafety.ALLOWED_IMPORTS.containsKey(importType)) {
+                Type givenFallbackType =
+                        Type.primitive(PrimitiveType.valueOf(ValidImportsForSafety.ALLOWED_IMPORTS.get(importType)));
+                if (givenFallbackType.equals(value.getFallback())) {
+                    return Stream.empty();
+                } else {
+                    return Stream.of(String.format(
+                            "Mismatched base type. %s must have a base type of %s in order to declare safety.",
```

Comment:
I'm not convinced that this is the best place for this check - as a user, I might be confused if I suddenly start getting a mismatched fallback type error when adding safety annotations. We could just skip this check, unless there's particular harm in allowing safety annotations for external imports with an invalid fallback type.

Reply:
@nathklei you suggested this check - do you think we absolutely need it?